### PR TITLE
fix(@angular-devkit/build-ng-packagr): add tsickle as optional peer dep

### PR DESCRIPTION
**Pull request by filipesilva**
This way it'll be automatically updated if present.

### CODE REVIEW

**Comment by IgorMinar** [comment]
this is cool @filipesilva! I didn't know that npm and yarn now supported optional peerDeps.

**Comment by IgorMinar** [comment]
btw I came across some issues with tsickle 0.37.1 which we might need to be aware of: https://github.com/angular/angular/pull/33788

**Comment by dgp1130** [comment]
@IgorMinar, is this safe to merge or are those issues blocking for this PR as well?

**Comment by filipesilva** [comment]
@IgorMinar npm added support in https://github.com/npm/cli/pull/224 and yarn added in https://github.com/yarnpkg/yarn/issues/6487.

File: packages/angular_devkit/build_ng_packagr/package.json [hunk 1]
```diff
@@ -11,13 +11,19 @@
     "rxjs": "6.5.3"
   },
   "peerDependencies": {
-    "ng-packagr": "^9.0.0-rc.0"
+    "ng-packagr": "^9.0.0-rc.0",
+    "tsickle": "~0.37.1"
```
Comment (alan-agius4) [inline packages/angular_devkit/build_ng_packagr/package.json]: I was under the impression that the fix for this was the include the dependency in the packageGroup.

Comment (filipesilva) [reply packages/angular_devkit/build_ng_packagr/package.json]: Yeah that was the original idea, but then Charles brough the optional peer dep up that both ensures it is updated, and ensures that if it is present on install (outside of update) it should be the right version. Do you think there are problems with this approach?

Comment (alan-agius4) [reply packages/angular_devkit/build_ng_packagr/package.json]: Not at all, I was just curios to know why the change :)

**Comment by clydin** [comment]
Also of note, `npm` had a defect that was fixed in 6.13.0 (currently the latest release) that would cause an incorrect warning about the peer dependency being missing but only on the first install.
It was originally thought to be a registry server only issue.  However, upon fixing the npm registry it was found that one of npm's dependencies also needed a fix as well.  But as of today yarn, npm, and pnpm all support optional peer dependencies.

**Comment by filipesilva** [comment]
Discussed in a meeting, it's safe to do this because we will be removing it right after during the migration. This will just correct the peerdep failure if any exists.

**Comment by dgp1130** [comment]
Merged to 9.0.x patch: https://github.com/angular/angular-cli/commits/9.0.x.

**Pull request merged**